gitfame/_gitfame.py

```diff
@@ -472,11 +472,14 @@ def main(args=None):
     log.debug(args)
     if args.manpath is not None:
         import sys
-        from os import path
+        from os import fspath, path
         from shutil import copyfile
 
-        from pkg_resources import resource_filename
-        fi = resource_filename(__name__, 'git-fame.1')
+        try:  # py<3.9
+            import importlib_resources as resources
+        except ImportError:
+            from importlib import resources
+        fi = fspath(resources.files('gitfame').resolve() / 'git-fame.1')
```

Comment:
`fspath` isn't doing what you're hoping for here.

All `fspath` does is convert a path-like object to a string. In an on-disk mode, `resources.files()` returns a `pathlib.Path` object, so `fspath` will work as you expect, but if someone were to bundle this project as a zipapp, `resources.files()` would return a `zipfile.Path` (or `zipp.Path` object).  In general, `resources.files()` returns a [`Traversable`](https://docs.python.org/3/library/importlib.resources.abc.html#importlib.resources.abc.Traversable), which may or may not have a direct file system representation.

If what you want is a file on the file system, you can use [`importlib.resources.as_file`](https://docs.python.org/3/library/importlib.resources.html#importlib.resources.as_file), a context manager that will ensure the contents are on the disk (if not already) and then clean them up when the context exits (regardless of whether the resources were stored in a zip file or in a database or whatever).

But since all you're doing is copying the contents, you'd be much better off to directly write the contents:

```
fo = pathlib.Path(args.manpath, 'git-fame.1')
fo.write_bytes(resources.files(__name__).joinpath('git-fame.1').read_bytes())
log.info("written:%s", fo)
```

Reply:
ah nice; I'll have to update a few other projects too